magicctapipe/io/gadf.py

```diff
@@ -0,0 +1,312 @@
+    -------
+    event_hdu: astropy.io.fits.hdu.table.BinTableHDU
+        Event HDU
+    """
+
+    mjdreff, mjdrefi = np.modf(MJDREF.mjd)
+
+    time_start = Time(event_data["timestamp"][0], format="unix", scale="utc")
+    time_start_iso = time_start.to_value("iso", "date_hms")
+
+    time_end = Time(event_data["timestamp"][-1], format="unix", scale="utc")
+    time_end_iso = time_end.to_value("iso", "date_hms")
+
+    elapsed_time = time_end - time_start
+    effective_time = on_time * deadc
+
+    event_coords = SkyCoord(
+        ra=event_data["reco_ra"], dec=event_data["reco_dec"], frame="icrs"
+    )
+
+    event_coords = event_coords.galactic
+
+    try:
+        # Try to get the coordinate from the source name
+        source_coord = SkyCoord.from_name(source_name, frame="icrs")
+
+    except Exception:
+        # Use the input RA/Dec coordinate instead
+        if (source_ra is None) or (source_dec is None):
+            raise ValueError(
+                f"The source name '{source_name}' could not be resolved, "
+                "but the input RA/Dec coordinate is also set to None."
+            )
+        else:
+            logger.warning(
+                f"WARNING: The source name '{source_name}' could not be resolved. "
+                f"Setting the input RA/Dec coordinate ({source_ra}, {source_dec})..."
+            )
+            source_coord = SkyCoord(ra=source_ra, dec=source_dec, frame="icrs")
+
+    # Create a table
+    qtable = QTable(
+        data={
+            "EVENT_ID": event_data["event_id"],
+            "TIME": event_data["timestamp"],
+            "RA": event_data["reco_ra"],
+            "DEC": event_data["reco_dec"],
+            "ENERGY": event_data["reco_energy"],
+            "GAMMANESS": event_data["gammaness"],
+            "MULTIP": event_data["multiplicity"],
+            "GLON": event_coords.l.to(u.deg),
+            "GLAT": event_coords.b.to(u.deg),
+            "ALT": event_data["reco_alt"].to(u.deg),
+            "AZ": event_data["reco_az"].to(u.deg),
+        }
+    )
+
+    # Create a header
+    header = fits.Header(
+        cards=[
+            ("CREATED", Time.now().utc.iso),
+            ("HDUCLAS1", "EVENTS"),
+            ("OBS_ID", np.unique(event_data["obs_id"])[0]),
+            ("DATE-OBS", time_start_iso[:10]),
+            ("TIME-OBS", time_start_iso[11:]),
+            ("DATE-END", time_end_iso[:10]),
+            ("TIME-END", time_end_iso[11:]),
+            ("TSTART", time_start.value),
+            ("TSTOP", time_end.value),
+            ("MJDREFI", mjdrefi),
+            ("MJDREFF", mjdreff),
+            ("TIMEUNIT", "s"),
+            ("TIMESYS", "UTC"),
+            ("TIMEREF", "TOPOCENTER"),
+            ("ONTIME", on_time.value),
+            ("TELAPSE", elapsed_time.to_value(u.s)),
+            ("DEADC", deadc),
+            ("LIVETIME", effective_time.value),
+            ("OBJECT", source_name),
+            ("OBS_MODE", "WOBBLE"),
+            ("N_TELS", 3),
```

Comment:
I guess this funciton can be also executed for MAGIC-only analysis, in which case this and next two lines would not be correct. 
I suppose that getting max of multiplicity could be included here. The chance that we have no 3-tel events in a run should be very small. 
More difficult to fill in the next two lines properly, because 2 telescopes can mean MAGIC-only, but also the recent M2+LST1 data

Reply:
Thanks for your comment, indeed the keys are wrong in case of 2 tels observations. I changed the script to use the maximum multiplicity for `N_TELS` as you suggested. And for `TELLIST` and `INSTRUME`, I changed the script to assign proper instruments using the `combo_types` parameter. Now the instrument will be "LST1_M1_M2", "LST1_M1", "LST1_M2" or "M1_M2" depending on the data. It would be great if you could check the updated lines.

Reply:
thanks, but where did you change it? this part still says 3 tels

Reply:
Sorry, I have not yet pushed the commits, since I have not yet implemented your comments. But anyway let me push the commits now that I implemented so far.

Reply:
Pushed them now.